core/domain/user_services_test.py

```diff
@@ -1318,6 +1603,242 @@ def test_record_user_started_state_translation_tutorial(self):
 
+    def test_get_human_readable_user_ids(self):
+        auth_ids = ['user1', 'user2', 'user3']
+        user_emails = [
+            '1user@example.com',
+            '2user@example.com',
+            '3user@example.com']
+        user_ids = []
+
+        for i, auth_id in enumerate(auth_ids):
+            user_ids.append(user_services.create_new_user(
+                auth_id, user_emails[i]).user_id)
+
+        user_services.set_username(user_ids[0], 'username1')
+        user_services.mark_user_for_deletion(user_ids[1])
+
+        user_settings3 = user_services.get_user_settings(user_ids[2])
+
+        usernames = [
+            'username1',
+            user_services.LABEL_FOR_USER_BEING_DELETED,
+            (
+                '[Awaiting user registration: %s]' %
+                user_settings3.truncated_email
+            )
+        ]
+
+        self.assertEqual(
+            usernames,
+            user_services.get_human_readable_user_ids(user_ids)
+        )
+
+    def test_get_human_readable_user_ids_no_user_raises_error(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        with self.assertRaisesRegex(Exception, 'User not found.'):
+            user_services.get_human_readable_user_ids([user_id, 'another_id'])
+
+    def test_record_user_started_state_editor_tutorial(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.record_user_started_state_editor_tutorial(user_id)
+        user_settings = user_services.get_user_settings(user_id)
+        orig = user_settings.last_started_state_editor_tutorial
+
+        user_services.record_user_started_state_editor_tutorial(user_id)
+        user_settings = user_services.get_user_settings(user_id)
+
+        self.assertGreaterEqual(
+            user_settings.last_started_state_editor_tutorial,
+            orig
+        )
+
+    def test_create_user_contributions_with_bot_user_id_returns_none(self):
+        user_id = feconf.MIGRATION_BOT_USER_ID
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+
+        user_contrib = user_services.create_user_contributions(
+            user_id,
+            created_exp_ids,
+            edited_exp_ids)
+
+        self.assertIsNone(user_contrib)
+
+    def test_create_user_contributions_already_existing_raises_error(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        error_msg = (
+            'User contributions model for user %s already exists.'
+            % user_id)
+
+        with self.assertRaisesRegex(Exception, error_msg):
```

Comment:
It's not quite clear that the model already exists from reading the test; this needs more justification/explanation.

Reply:
I don't see an explanation?

Reply:
I added comments that explain why contributions exist. Please let me know if I should add anything else!